pyflow/core/add_newblock_button.py

```diff
@@ -0,0 +1,82 @@
+# Pyflow an open-source tool for modular visual programing in python
+# Copyright (C) 2021-2022 Bycelium <https://www.gnu.org/licenses/>
+
+""" Module for the button to create a linked block."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional
+
+from PyQt5.QtCore import QRectF, QPoint
+from PyQt5.QtGui import QBrush, QColor, QPainter, QPen, QPolygon
+from PyQt5.QtWidgets import QGraphicsItem, QStyleOptionGraphicsItem, QWidget
+
+if TYPE_CHECKING:
+    from PyQt5.QtWidgets import QGraphicsSceneHoverEvent
+    from pyflow.blocks.executableblock import ExecutableBlock
+    from pyflow.core.edge import Edge
+
+
+class AddNewBlockButton(QGraphicsItem):
```

Comment:
Maybe this and AddEdgeButton should inherit from the same object to reduce duplication